src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -1129,6 +1129,12 @@ pub fn rustc_cargo_env(
         cargo.env("CFG_DEFAULT_LINKER", s);
     }
 
+    // Enable rustc's cfgs for `rust-lld` when requested.
+    if builder.config.lld_enabled {
+        cargo.rustflag("--cfg=use_rust_lld");
+        cargo.rustdocflag("--cfg=use_rust_lld");
```

Comment:
I think using an environment variable (e.g., like CFG_DEFAULT_LINKER just above) is better. That way you also get free rebuilds that are localized to that one library when tweaking this rather than the full compiler tree.

Reply:
Good idea. I'm not sure I have the same use-cases in mind in you when you say "tweaking this" so I've pushed 1978bc22112ec7d0aa5fbe627c8f95d4d3d4c921 to address this. Let me know if that matches your expectations?

Reply:
Yeah I just meant toggling the default would lead to a world-rebuild with a --cfg and only a rebuild of the dependencies of the use site otherwise.

Reply:
Yeah, ok then I believe these 2 commits should resolve this.